pkg/storage/indexheader/header_test.go

```diff
@@ -177,15 +179,31 @@ func Test_DownsampleSparseIndexHeader(t *testing.T) {
 			// label names are equal between original and downsampled sparse index-headers
 			require.ElementsMatch(t, downsampleLabelNames, origLabelNames)
 
-			// downsampled postings are a subset of the original sparse index-header postings
 			origIdxpbTbl := br1.postingsOffsetTable.NewSparsePostingOffsetTable()
 			downsampleIdxpbTbl := br2.postingsOffsetTable.NewSparsePostingOffsetTable()
 
 			step := tt.inMemSamplingRate / tt.protoRate
 			for name, vals := range origIdxpbTbl.Postings {
 				downsampledOffsets := downsampleIdxpbTbl.Postings[name].Offsets
-				require.Equal(t, (len(vals.Offsets)+step-1)/step, len(downsampledOffsets))
-				require.Subset(t, vals.Offsets, downsampledOffsets)
+
+				// downsampled postings are a subset of the original sparse index-header postings
+				if (tt.inMemSamplingRate > tt.protoRate) && (tt.inMemSamplingRate%tt.protoRate == 0) {
+					require.Equal(t, (len(vals.Offsets)+step-1)/step, len(downsampledOffsets))
+					require.Subset(t, vals.Offsets, downsampledOffsets)
+				}
+
+				// cannot "upsample" a sparse index header. When the sampling value found in sparse header is larger
+				// than the configured rate, should rebuild sparse headers. The rebuilt headers should be at least
+				// as large as the original sparse index header postings.
+				if tt.inMemSamplingRate < tt.protoRate {
```

Comment:
since you moved downsampling to the postingsOffseTable, do you want to make the test simpler and just test this in index/postings_test.go too or prefer to keep it here?